### PR TITLE
Some miri cleanups

### PR DESCRIPTION
**Pull request by oli-obk**
r? @eddyb

### CODE REVIEW

File: src/librustc_mir/interpret/const_eval.rs [hunk 1]
```diff
@@ -20,77 +20,76 @@ use rustc_const_math::ConstInt;
...
 
 pub fn eval_body_as_integer<'a, 'tcx>(
     tcx: TyCtxt<'a, 'tcx, 'tcx>,
     param_env: ty::ParamEnv<'tcx>,
     instance: Instance<'tcx>,
 ) -> EvalResult<'tcx, ConstInt> {
-    let (ptr_ty, ecx) = eval_body(tcx, instance, param_env);
+    let ptr_ty = eval_body(tcx, instance, param_env);
     let (ptr, ty) = ptr_ty?;
+    let ecx = mk_eval_cx(tcx, instance.substs, param_env);
```
Comment (eddyb) [inline src/librustc_mir/interpret/const_eval.rs]: Is the new `substs` field on `EvalContext` just for this?

Comment (oli-obk) [reply src/librustc_mir/interpret/const_eval.rs]: jup. We used to push a stackframe just to have substs.

Comment (oli-obk) [reply src/librustc_mir/interpret/const_eval.rs]: I removed the substs part and just simplified the interface

File: src/librustc_mir/interpret/const_eval.rs [hunk 2]
```diff
@@ -340,20 +350,22 @@ pub fn const_eval_provider<'a, 'tcx>(
...
                 let () = unwrap_miri(&ecx, Err(err));
                 Ok(ok)
             },
-            ((Ok(_), _), Err(err)) => {
+            (Ok(_), Err(err)) => {
                 Err(err)
             },
-            ((Err(_), _), Err(err)) => Err(err),
-            ((Ok((miri_val, miri_ty)), mut ecx), Ok(ctfe)) => {
+            (Err(_), Err(err)) => Err(err),
```
Comment (kennytm) [inline src/librustc_mir/interpret/const_eval.rs]: Heh strange. Why these two arms are not merged 😄

**Review by eddyb — APPROVED** [review]

**Comment by eddyb** [comment]
@bors delegate=oli-obk

r=me with @kennytm's nit fixed

File: src/librustc_mir/interpret/const_eval.rs (hunk 2 above)
Comment (oli-obk) [reply src/librustc_mir/interpret/const_eval.rs]: Because the next commit will unmerge them because it's going to process the miri result ;)

**Pull request merged**